src/AST/Entities.dfy

```diff
@@ -0,0 +1,644 @@
+/// Dafny's modules, types, and members hierarchy
+/// =============================================
+
+include "Names.dfy"
+include "Syntax.dfy"
+include "../Utils/Library.dfy"
+include "../Utils/Lib.Sort.dfy"
+
+module {:options "-functionSyntax:4"} Bootstrap.AST.Entities
+  // Hierarchies of Dafny entities.
+  // See </doc/design/entities.md>.
+{
+  import opened Names
+  import opened Syntax.Exprs
+  import opened Syntax.Types
+  import ST = Syntax.Types
+  import opened Utils.Lib.Datatypes
+  import Utils.Lib.SetSort
+  import OS = Utils.Lib.Outcome.OfSeq
+
+  // DISCUSS: Should this module be parameterized by `TExpr`?
```

Comment:
I could see value in doing so, but I'd be inclined to only change it if we have a concrete use case, along with the change to implement that use case.